### DISCUSSION TITLE
Change the name of a field in User : for exemple "id" to "uid"

### DISCUSSION THREAD

**Discussion by mlisthenewcool** [fastapi-users/fastapi-users#1245] [Q&A]
Hello there,
Is it possible to change the name of a field in User ? I would love to change specifically the "id" field. 

I didn't investigate really deeply into the code but it seems it could be done on two levels :
- Directly into the DB, thus it should be also reflected in schemas and classes.
- Only in schemas.

Any ideas ?

**Comment by frankie567** [comment]
Unfortunately, changing the name of a pre-defined field is not possible, as lot of logic relies on it. You may try to use [`alias` from Pydantic](https://docs.pydantic.dev/1.10/usage/models/#reserved-names) and [`hybrid_property` from SQLAlchemy](https://docs.sqlalchemy.org/en/20/orm/extensions/hybrid.html), but that'll probably be quite tricky.
